### PR TITLE
Feature/join send socket with room

### PR DESCRIPTION
**Pull request by vietdangquoc**
# Ticket
#644 

# General
- [ ] No unnecessary files, debug logs, or commented code included.
- [ ] PR has at least 1 peer approval before senior review.
- [ ] I fully understand the code I submitted.
- [ ] No hardcoded configuration or secrets.
- [ ] **Environment Variables** changes? (Leave unchecked if N/A)

# Design & Solution Clarity
- [ ] A design diagram (sequence / flow) is attached or linked.
- [ ] The implementation matches the design.
- [ ] All major flows and decision points are represented in the diagram.
- [ ] Error handling and edge cases are clearly defined in the design.

Design link (diagram / doc):
`<paste link here>`

# Testing
- [ ] Unit tests cover main logic and important edge cases.

> Note: Complex logic must not rely on manual testing only.

# UI evidence (For frontend only)
`<paste screenshots here>`

# Peer Review Checklist
- [ ] I understand the overall solution and flow.
- [ ] The code follows the provided design diagram.
- [ ] No obvious logic or security issues found.
- [ ] Tests are sufficient for the scope of this change.

### CODE REVIEW

File: dong-service/scheduler/cancel_expired_orders_job.go [hunk 1]
```diff
@@ -45,10 +43,6 @@ func (j *CancelExpiredOrdersJob) Run(ctx context.Context) error {
 	if count > 0 {
 		logger.Info().Int64("count", count).Msg("Cancelled expired orders")
```
Comment (hoangdohuy) [inline dong-service/scheduler/cancel_expired_orders_job.go]: sao xóa đoạn này nhỉ?

File: socket-service/handlers/http_handler.go [hunk 2]
```diff
@@ -30,24 +33,71 @@ func (h *HTTPHandler) SaveEvent(c *gin.Context) {
...
 			} else {
-				logger.Info().Msgf("Event sent to online user %s", event.ReceiveAddress)
 				sentToOnline = true
+				sentMap[conn] = struct{}{}
 			}
 		}
 	}
-    if (!sentToOnline) {
+
+	if roomConns, ok := h.wsSvc.GetRoomConnections(event.ReceiveAddress); ok {
```
[the remark this answers is not in the document]
Comment (vietdangquoc) [reply socket-service/handlers/http_handler.go]: nếu onliine -> add vào [] -> send socket

File: socket-service/handlers/ws_handler.go [hunk 3]
```diff
@@ -110,12 +118,37 @@ func (h *WSHandler) HandleWS(c *gin.Context) {
...
+				logger.Info().Msgf("Heartbeat check received from user %s", userAddress)
+				conn.SetWriteDeadline(time.Now().Add(time.Duration(h.cfg.WebSocket.WriteWait) * time.Second))
+				conn.WriteMessage(websocket.TextMessage, []byte(constant.HeartbeatAck))
+				logger.Info().Msgf("Heartbeat ack sent to user %s", userAddress)
+				continue
+			}
+			// Handle join_room/leave_room
+			type RoomMsg struct {
+				Type string `json:"type"`
+				Room string `json:"room"`
```
Comment (hoangdohuy) [inline socket-service/handlers/ws_handler.go]: validate hoặc bỏ
